wp/lib/bap_wp/src/constraint.ml

```diff
@@ -310,7 +345,9 @@ let get_refuted_goals ?filter_out:(filter_out = []) (constr : t)
             Seq.append (worker c current_path current_registers olds news) accum)
     | Subst (e, o, n) ->
       let n' = List.map n ~f:(fun x -> Expr.substitute x olds news |> eval_model_exn model) in
-      worker e current_path current_registers (olds @ o) (news @ n')
+      let oldsnews = List.zip_exn olds news in
```

Comment:
Do we generate list sizes of 1 a lot like the case in `eval` or does `get_refuted_goals` not need that special case?

Reply:
That's a good point. I think they are using the same Constr.t objects, so I expect also this one to include almost exclusively lists of size 1.

Reply:
It appears n-queens-7 goes from ~15s to ~14s by replicating the list logic here. Removing the fancy logic from eval in turn takes the total runtime from 15s to 21.5s

Reply:
A majority of the time is still in the precondition constraint evaluation phase and get_refuted_goals is relatively quick.